src/sqlparser/src/ast/statement.rs

```diff
@@ -634,6 +634,145 @@ impl fmt::Display for CreateSubscriptionStatement {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum CursorFrom {
+    Query(Box<Query>),
+    Subscription(ObjectName, Since),
```

Comment:
I think it is clearer to use `Option<Since>` here and avoid introducing `WithSnapshot` in the `Since` enum because there is not `Since` keyword in the statement in this case.